magic/oracle.py

```diff
@@ -231,3 +231,24 @@ def insert_scryfall_card(sfcard):
         multiverse.insert_card(c)
     multiverse.update_cache()
     CARDS_BY_NAME[sfcard['name']] = load_card(sfcard['name'])
+
+def last_pd_rotation_changes():
+    current_code = rotation.last_rotation_ex()['code']
+    previous = multiverse.SEASONS[multiverse.SEASONS.index(current_code) - 1]
+    previous_id = multiverse.get_format_id("Penny Dreadful {f}".format(f=previous))
+    current_id = multiverse.get_format_id("Penny Dreadful")
+    return changes_between_formats(previous_id, current_id)
+
+def changes_between_formats(f1, f2):
+    return [query_diff_formats(f2, f1), query_diff_formats(f1, f2)]
+
+def query_diff_formats(f1, f2):
+    query = 'c.id IN (SELECT card_id FROM card_legality WHERE format_id = {format1}) AND c.id NOT IN (SELECT card_id FROM card_legality WHERE format_id = {format2})'
+    removals_query = multiverse.base_query(where=query.format(format1=f1, format2=f2))
+
+    sql = """
```

Comment:
This line seems redundant. If sql = base_query and base_query = removals_query then you only need one variable, not three? It was probably more complicated than this in some earlier version?

Reply:
Any reason you're using base_query and not cached_base_query?

Reply:
Oh, that's important!

Reply:
No reason other than ignorance. I'll check that.